libraries/entities/src/EntityItem.cpp

```diff
@@ -780,7 +780,24 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
     READ_ENTITY_PROPERTY(PROP_LIFETIME, float, updateLifetime);
     READ_ENTITY_PROPERTY(PROP_SCRIPT, QString, setScript);
     READ_ENTITY_PROPERTY(PROP_SCRIPT_TIMESTAMP, quint64, setScriptTimestamp);
+
+    bool previousOverwriteLocalDataValue = overwriteLocalData;
+
+    if (!overwriteLocalData) {
+        // We've decided not to read the data in this update because we have a newer local state of the entity.
+        // In order to work around a bug stopping server script changes from being received by an entity script server
+        // running a script that continously updates an entity, we force the overwriting of the server scripts
+        // if we believe this update might have a newer version than the one we have locally.
+
+        if (_lastEditedFromRemote > _serverScriptsChangedTimestamp) {
+            overwriteLocalData = true;
+        }
+    }
+
     READ_ENTITY_PROPERTY(PROP_SERVER_SCRIPTS, QString, setServerScripts);
+
+    overwriteLocalData = previousOverwriteLocalDataValue;
```

Comment:
Let's try this instead...
1) change the comment to speak to the actual mechanism involved here, basically you're saying "always allow more recent changes to the `_serverScripts` property even when local changes to other properties are more recent.
2) Scoping is your friend, you don't need to to remember/set/reset `overwriteLocalData` just use a new narrower scope...


```
 +    // Always honor newer changes in PROP_SERVER_SCRIPTS even when local changes to other properties
 +    // would have caused us to ignore this change as less recent than our local changes.
 +    // This allows remote clients to change the PROP_SERVER_SCRIPTS when the entity script server is
 +    // running a script that continuously updates an entity.
 +    {
 +        bool overwriteLocalData = (_lastEditedFromRemote > _serverScriptsChangedTimestamp);
 +
           READ_ENTITY_PROPERTY(PROP_SERVER_SCRIPTS, QString, setServerScripts);
 +    }
```